src/plugins/discover/public/application/main/utils/update_saved_search.ts

```diff
@@ -38,9 +39,11 @@ export function updateSavedSearch({
   services: DiscoverServices;
   useFilterAndQueryServices?: boolean;
 }) {
-  if (dataView) {
+  if (dataView && savedSearch.searchSource.getField('index')?.id !== dataView.id) {
     savedSearch.searchSource.setField('index', dataView);
-    savedSearch.usesAdHocDataView = !dataView.isPersisted();
+    if (!dataView.isPersisted()) {
+      savedSearch.usesAdHocDataView = true;
+    }
```

Comment:
I'm curious about these changes. Why do we compare the data view IDs now, and only conditionally add `usesAdHocDataView`?

Reply:
This is a change that we could remove from this PR or do it in another one, for better scoping. 
It's about removing redundancy, there's no reason so set `index` again, if it hasn't changed. and why setting `usesAdHocDataView` to false, when it's just ab out to detect if the saved search is using an ad-hoc data view, and undefined equals false in this case. 

So this changes are optional to this PR

Reply:
Ok, I think I get what you mean now. So you're basically avoiding persisting any value for `usesAdHocDataView` to the SO if it's not an ad hoc data view? In any case, I think it's fine to keep these in this PR.